src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -689,9 +714,13 @@ def _deploy_arm_template_at_tenant_scope(cmd,
     mgmt_client = _get_deployment_management_client(cmd.cli_ctx, plug_pipeline=(template_uri is None and template_spec is None))
 
     if cmd.supported_api_version(min_api='2019-10-01', resource_type=ResourceType.MGMT_RESOURCE_RESOURCES):
+        from msrestazure.azure_exceptions import CloudError
         ScopedDeployment = cmd.get_models('ScopedDeployment')
         deployment = ScopedDeployment(properties=deployment_properties, location=deployment_location)
-        validation_poller = mgmt_client.validate_at_tenant_scope(deployment_name=deployment_name, parameters=deployment)
+        try:
+            validation_poller = mgmt_client.validate_at_tenant_scope(deployment_name=deployment_name, parameters=deployment)
+        except CloudError as cx:
```

Comment:
could you still catch it when track2 SDK is adopted?

Reply:
I guess catch the `HttpResponseError` may be needed, because the processing of `status==400` by the track 2 SDK still does not throw an exception, but returns the result of `DeploymentValidateResult`.  [code link](https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/resources/azure-mgmt-resource/azure/mgmt/resource/resources/v2020_06_01/operations/_deployments_operations.py#L1158)

Then, HttpResponseError may be thrown later because ARMPolling (from azure.mgmt.core) is called later:
```
def _raise_if_bad_http_status_and_method(response):
    # type: (ResponseType) -> None
    """Check response status code is valid.

    Must be 200, 201, 202, or 204.

    :raises: BadStatus if invalid status.
    """
    code = response.status_code
    if code in {200, 201, 202, 204}:
        return
    raise BadStatus(
        "Invalid return status {!r} for {!r} operation".format(
            code, response.request.method
        )
    )

def initialize(self, client, initial_response, deserialization_callback):

    ...

    try:
        _raise_if_bad_http_status_and_method(self._initial_response.http_response)
        self._status = self._operation.set_initial_status(initial_response)

    except BadStatus as err:
        self._status = "Failed"
        raise HttpResponseError(response=initial_response.http_response, error=err)
```

I will verify it when I migrate track 2 later. If there is still this problem, we can consider giving suggestions to the Python SDK of track 2 to throw exceptions in this case of `status==400`, right?

Reply:
If you have tests about the exception type, it will be easy to catch the change.

Reply:
Yes, however there are some breaking changes during the initialization of `ResourceManagementClient` when migrating to track 2, which needs further investigation~